src/colorizer/canvas/elements/scalebar.ts

```diff
@@ -0,0 +1,131 @@
+import { Vector2 } from "three";
+
+import { numberToSciNotation } from "../../utils/math_utils";
+import { BaseRenderParams, defaultStyleOptions, EMPTY_RENDER_INFO, FontStyleOptions, RenderInfo } from "../types";
+import { configureCanvasText, getPixelRatio, renderCanvasText } from "../utils";
+
+export type ScaleBarOptions = FontStyleOptions & {
+  minWidthPx: number;
+  visible: boolean;
+};
+
+export type ScaleBarParams = BaseRenderParams & {
+  frameSizeInCanvasCoordinates: Vector2;
+};
```

Comment:
Okay, weird quirk I want to point out here. In `header` and `footer`, the visibility is passed in through `params`, while in the `scalebar` and `timestamp` it's passed in via `options`. I set it up differently because the latter can be a simple on/off toggle and should be controlled from `CanvasWrapper`, while the former is dependent on other canvas states (e.g. whether exporting is occurring) and should not be controllable from outside `CanvasWithOverlay`. 

Let me know if this seems particularly egregious?